transport/httptransport/http_transport.go

```diff
@@ -61,6 +96,7 @@ func (h *httpTransport) Execute(ctx context.Context, transportInfo []byte, dealI
 	tctx, cancel := context.WithCancel(ctx)
 	t := &transfer{
 		cancel:         cancel,
+		h:              h,
```

Comment:
nit: can we just store the variables on a backoff.Backoff object, and pass that object here instead of passing the httpTransport pointer?

Reply:
done.